### PR TITLE
feat-range-parser

### PR DESCRIPTION
**Pull request by lohanidamodar**
Adds content range header parser and relevant getters to get the content range start, content range end, content range size and content range unit

### CODE REVIEW

File: tests/RequestTest.php [hunk 1]
```diff
@@ -196,4 +196,14 @@ public function testGetAccept()
         $this->assertEquals('accept', $this->request->getAccept('default'));
     }
 
+    public function testContentRange()
+    {
```
Comment (eldadfux) [inline tests/RequestTest.php:200]: Let's add more assertions, including ones that should fail.

**Review by eldadfux — COMMENTED** [review]
Would you like to add support for the range header as well? does it have the same syntax?

File: tests/RequestTest.php (hunk 1 above)
Comment (lohanidamodar) [reply tests/RequestTest.php:200]: done

File: tests/RequestTest.php [hunk 2]
```diff
@@ -196,4 +196,30 @@ public function testGetAccept()
...
+
+        $_SERVER['HTTP_CONTENT_RANGE'] = " 0-499/2000";
+        
+        $this->request = new Request();
+        $this->assertEquals(null, $this->request->getContentRangeUnit());
+        $this->assertEquals(null, $this->request->getContentRangeStart());
+        $this->assertEquals(null, $this->request->getContentRangeEnd());
+        $this->assertEquals(null, $this->request->getContentRangeSize());
+
+        $_SERVER['HTTP_CONTENT_RANGE'] = "bytes 0-499/";
```
Comment (TorstenDittmann) [inline tests/RequestTest.php:216]: Does it make sense to test headers that doesn't resemble the standarts?

- `bytes 0--499/2000`
- `bytes 0-499/20/00`
- `bytes 0-499test/2000`
- `bytes 0-49.9/200.0`
- `bytes 0-49,9/200,0`

**Review by TorstenDittmann — APPROVED** [review]
Looks good beside the small comment I had 🙂

**Comment by lohanidamodar** [comment]
> Would you like to add support for the range header as well? does it have the same syntax?

It has totally different format and I'll add it later in separate PR

File: tests/RequestTest.php (hunk 2 above)
Comment (lohanidamodar) [reply tests/RequestTest.php:216]: May be I should thanks for the ideas

**Comment by eldadfux** [comment]
I can merge this and release a version once we have the new assertions.

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**